src/bucket.ts

```diff
@@ -3174,6 +3185,201 @@ class Bucket extends ServiceObject {
+   * @param {string} [options.userProject] The ID of the project which will be
+   *     billed for the request.
+   * @param {string|boolean} [options.validation] Possible values: `"md5"`,
+   *     `"crc32c"`, or `false`. By default, data integrity is validated with an
+   *     MD5 checksum for maximum reliability. CRC32c will provide better
+   *     performance with less reliability. You may also choose to skip
+   * validation completely, however this is **not recommended**.
+   */
+  /**
+   * @typedef {array} UploadDirectoryResponse
+   * @property {object} [response] Response for each of the uploaded files.
+   * @param {string} [filename] Absolute file path.
+   * @param {string | Error} [status] Status of the file upload. "success"
+   *     or an Error returned while attempting to upload current file.
+   */
+  /**
+   * @callback UploadDirectoryCallback
+   * @param {?Error} err Request error, if any.
+   * @param {array} resp The list of object responses for every uploaded file.
+   */
+
+  /**
+   * Upload a directory to the bucket. This is a convenience method for multiple
+   * {@link Bucket#upload}.
+   *
+   * You can specify whether or not an upload is resumable by setting
+   * `options.resumable`. *Resumable uploads are enabled by default if your
+   * input file is larger than 5 MB.*
+   *
+   * For faster crc32c computation, you must manually install
+   * [`fast-crc32c`](http://www.gitnpm.com/fast-crc32c):
+   *
+   *     $ npm install --save fast-crc32c
+   *
+   * @see [Upload Options (Simple or Resumable)]{@link https://cloud.google.com/storage/docs/json_api/v1/how-tos/upload#uploads}
+   * @see [Objects: insert API Documentation]{@link https://cloud.google.com/storage/docs/json_api/v1/objects/insert}
+   *
+   * @param {string} directoryPathString The fully qualified path to the
+   *     directory you wish to upload to your bucket.
+   * @param {UploadDirectoryOptions} [options] Configuration options.
+   * @param {UploadDirectoryCallback} [callback] Callback function.
+   * @returns {Promise<UploadDirectoryResponse>}
+   *
+   * @example
+   * const {Storage} = require('@google-cloud/storage');
+   * const storage = new Storage();
+   * const bucket = storage.bucket('albums');
+   *
+   * //-
+   * // Upload a directory from a local path.
+   * //-
+   * bucket.uploadDirectory('/local/path/myfolder', function(err, resp) {
+   *   // Your bucket now contains:
+   *   // - "myfolder/" directory (with the contents of `/local/path/myfolder/')
+   *
+   *   // `file` is an instance of a File object that refers to your new file.
+   * });
+   *
+   *
+   * //-
+   * // You can also have files gzip'd on the fly.
+   * //-
+   * bucket.uploadDirectory(path.join(__dirname, 'index', { gzip: true }, function(err, resp) {
+   *   // Your bucket now contains:
+   *   // - "index/" directory (automatically compressed with gzip)
+   *
+   *   // Downloading the file with `file.download` will automatically decode
+   * the
```

Comment:
looks like this comment needs some formatting love, also I'm a little confused about the context of `file.download`, are we saying any of the files you just uploaded will be automatically decoded?

Reply:
Yes, the client library always requests data from the server "as is" via the [headers](https://github.com/googleapis/nodejs-storage/blob/master/src/file.ts#L1238-L1241) ([docs](https://cloud.google.com/storage/docs/transcoding#decompressive_transcoding)) and automatically decompresses if [`isCompressed`](https://github.com/googleapis/nodejs-storage/blob/master/src/file.ts#L1306)